src/BadgeRegistry.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { IBadgeRegistry } from "./interfaces/IBadgeRegistry.sol";
+
+/// @title Badge Registry
+/// @author KarmaGap | 0xneves.eth
+contract BadgeRegistry is IBadgeRegistry {
+  /// Map of badges registered in the registry
+  mapping(bytes32 => Badge) public badges;
+
+  /// @notice Registers a badge in the registry
+  /// @param badge Badge data struct
+  function create(Badge calldata badge) public returns (bytes32) {
+    bytes32 badgeId = generateId(badge);
+    if (bytes(badges[badgeId].name).length > 0) revert BadgeAlreadyRegistered();
```

Comment:
`if (bytes(badges[badgeId].name).length > 0)` is not foolproof. A badge with an empty name could be considered non-existent.

Reply:
That is exactly the idea, if the badge doesn't have a name them it should not exist